lhotse/dataset/sampling/round_robin.py

```diff
@@ -185,14 +195,16 @@ def _next_batch(self) -> Union[CutSet, Tuple[CutSet]]:
         return batch
 
     def _set_next_idx(self) -> None:
-        if self.randomize and len(self._nondepleted_samplers_indices) > 1:
-            self._cur_sampler_idx = self.rng.choice(
-                [
-                    i
-                    for i in range(len(self._nondepleted_samplers_indices))
-                    if i != self._cur_sampler_idx
-                ]
-            )
+        if self.randomize is not False and len(self._nondepleted_samplers_indices) > 1:
+            N = [
+                i
+                for i in range(len(self._nondepleted_samplers_indices))
+                if i != self._cur_sampler_idx
+            ]
+            p = [x for i, x in enumerate(self.randomize) if i != self._cur_sampler_idx]
```

Comment:
shouldn't you only select the probabilities for nondepleted samplers here?

Reply:
You're right. I have changed it now to allow selecting from all non-depleted samplers (including the current one).